MekHQ/src/mekhq/campaign/unit/Unit.java

```diff
@@ -1113,7 +1113,7 @@ else if (heatSinkType == MiscType.F_HEAT_SINK)
     }
 
     public long getSellValue() {
-        long partsValue = 0;
+        double partsValue = 0;
```

Comment:
I wish Java had a native decimal type we could use for currency. I'm not 100% sure `BigDecimal` is the right type either, perhaps Joda Money's `BigMoney` type? We've had a few bugs reported regarding costs being "off", which mostly seem like rounding issues.

Reply:
I think I'd rather the cost related work be broken out into a separate PR so we can perhaps explore better options (if any exist).

Reply:
I think this first pass is a good approach. Now we use double all the way in the code and only truncate at the end. The main issue right now is that we are using things like are doubles (like weights) with costs in long/int and some methods are casting to long in every single operation, introducing a lot of rounding issues.

With this at least the method signature is the same (long return), but internally we don't accumulate any rounding errors.